client/json-rpc/src/async_client/client.rs

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::async_client::{
+    defaults, types as jsonrpc, Error, JsonRpcResponse, RetryStrategy, State,
+    WaitForTransactionError,
+};
+use libra_crypto::hash::CryptoHash;
+use libra_types::{
+    account_address::AccountAddress,
+    transaction::{SignedTransaction, Transaction},
+};
+use serde::de::DeserializeOwned;
+use serde_json::json;
+use std::time::Duration;
+
+#[derive(Debug)]
+pub struct Response<R> {
+    pub result: R,
+    pub state: State,
+}
```

Comment:
I think we should `impl<R> Deref for Response<R>` that returns the interior. Then you can do

```
 let result = client.get_account(address).await?;
let version = result.state;
if let Some(acc) = *result {
  // ....
}
```

Reply:
interesting, let me check it out.